src/lxc/lxc_attach.c

```diff
@@ -273,7 +273,12 @@ static int get_pty_on_host(struct lxc_container *c, struct wrapargs *wrap, int *
 	if (lxc_console_create(conf) < 0)
 		return -1;
 	ts = conf->console.tty_state;
-	ts->stdoutfd = STDOUT_FILENO;
+	/* If stdout is not a tty but stderr is, we write to stderr. */
+	if (!isatty(STDOUT_FILENO) && isatty(STDERR_FILENO))
+		ts->stdoutfd = STDERR_FILENO;
+	else
+		ts->stdoutfd = STDOUT_FILENO;
+
```

Comment:
This does *not* redirect container's stdout to stderr. `sh -c 'echo ERR >&2'` is merely a synthetic reproducer command (running in the container) that writes something to stderr. But lxc-attach apparently now redirects stderr of the command in the container to stdout of lxc-attach under some conditions -- that's the regression. In case it's clearer, you can also use this one:

    lxc-attach -n a -- yes --foo > /dev/null

yes' error message about the unknown option also goes to stderr, so you should see it. But lxc-attach redirects it to stdout now, therefore hiding it. (It was still working correctly until rc1 at least)